plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt

```diff
@@ -64,7 +64,7 @@ internal data class RawResults(
     val markedAsIdentifiedFiles: List<MarkedAsIdentifiedFile>,
     val listIgnoredFiles: List<IgnoredFile>,
     val listPendingFiles: List<String>,
-    val listSnippets: Map<String, Set<Snippet>>,
+    val listSnippets: Map<String, Lazy<Set<Snippet>>>,
```

Comment:
IIUC the `LazzySet` is introduced to change the order of evaluating something, introducing the assumption that
the set is not accesses too early. I believe such magic must not be put into the scan result data structure, but rather made explicit. Would it be possible to make the order of processing explicit in the code, e.g. have one function which does things in the desired order. And finally once done, constructs the result without any lazy evaluated properties?

Reply:
I don't really understand your comment.
The lazy set is not introduced to manipulate the order of snippets in any way, it is here to load lazily the snippets at the last moment possible, to be able to load only the snippets as long as the limit has not been reached.

Reply:
I see, I guess I lack the full understanding. Can you help me understanding _why_ the "last moment possible" matter here? E.g. wouldn't a couple of moments earlier the limit also be known already?

Reply:
This would probably require a major rework, but would it make sense to do the mapping directly after the listing on a per-file basis? This would have the additional benefit that it could be done in parallel as well.

Reply:
I'm surprised that snippet choice is applied in the `scanner` stage. This implies that if you want a choice to 
take effect you need to re-scan. I suppose the turn-around time of making changes and seeing the effect is quite long. Given that, have there been any plans to move the application of the choices out of the scanner stage, or should there be? (If so, it'd mean that a solution for mapping snippets is needed, which works without knowing about the choices)

Reply:
@fviernau This is the current snippet choice workflow:

- The ORT pipeline is run
- The snippets are collected in the snippet report
- Based on this results, the snippet choices are added to `.ort.yml`
- An ORT pipeline is re-triggered: chosen snippets are hidden from the results and the snippet report 

If you want to apply the snippet choice without rescanning (e.g. in the Evaluator step ?), it means that every scanner run should _always_ list ALL snippets to allow the choice loop to occurs latter.

As I wrote in the commit message, we have customer monorepos with more than 400 000 snippets, and IMO embedding all these in the ORT Results is not scalable.

Reply:
Thanks for explaining this again. I think it's clearer to me now. Anyhow, I'd need to invest some time to deeper understand things for making a proper review. As I do not have a lot of time today, I step back from this review for now.